# HG changeset patch
# User Vytautas Janƒçauskas <unaudio@gmail.com>
# Date 1379626470 -10800
#      Fri Sep 20 00:34:30 2013 +0300
# Node ID 3954d835436ba3c91cb3d0137360c7fe27621fc9
# Parent  e36c9a2b291951b58484ac535711c7377bd2a820
doc: included all docstrings for the @audioplayer class in audio.txi as well as added additional explanations for the functionality

diff --git a/doc/interpreter/audio.txi b/doc/interpreter/audio.txi
--- a/doc/interpreter/audio.txi
+++ b/doc/interpreter/audio.txi
@@ -23,26 +23,74 @@
 @chapter Audio File Utilities
 
 @DOCSTRING(audioinfo)
 @DOCSTRING(audioread)
 @DOCSTRING(audiowrite)
 
 @chapter Audio Device Information
 
-@DOCSTRING(@audiodevinfo)
+@DOCSTRING(audiodevinfo)
 
 @chapter Audio Player
 
+The following methods are used to create and use audioplayer objects. These objects can be used to play back audio data stored in Octave matrices and arrays. The audioplayer object supports playback from various devices available to the system, blocking and non-blocking playback, convenient pausing and resuming and much more.
+
 @DOCSTRING(@audioplayer/audioplayer)
 
+@section Playback
+
+The following methods are used to control player playback.
+
+@DOCSTRING(@audioplayer/play)
+@DOCSTRING(@audioplayer/playblocking)
+@DOCSTRING(@audioplayer/pause)
+@DOCSTRING(@audioplayer/resume)
+@DOCSTRING(@audioplayer/stop)
+@DOCSTRING(@audioplayer/isplaying)
+
+@section Properties
+
+The remaining couple of methods are used to get and set various properties of the audioplayer object.
+
+@DOCSTRING(@audioplayer/get)
+@DOCSTRING(@audioplayer/set)
+
 @chapter Audio Recorder
 
+The following methods are used to create and use audiorecorder objects. These objects can be used to record audio data from various devices available to the system. You can use convenient methods to retrieve that data or audioplayer objects created from that data. Methods for blocking and non-blocking recording, pausing and resuming recording and much more is available.
+
 @DOCSTRING(@audiorecorder/audiorecorder)
 
+@section Recording
+
+The following methods control the recording process.
+
+@DOCSTRING(@audiorecorder/record)
+@DOCSTRING(@audiorecorder/recordblocking)
+@DOCSTRING(@audiorecorder/pause)
+@DOCSTRING(@audiorecorder/resume)
+@DOCSTRING(@audiorecorder/stop)
+@DOCSTRING(@audiorecorder/isrecording)
+
+@section Data Retrieval
+
+The following methods allow you to retrieve recorded audio data in various ways.
+
+@DOCSTRING(@audiorecorder/getaudiodata)
+@DOCSTRING(@audiorecorder/getplayer)
+@DOCSTRING(@audiorecorder/play)
+
+@section Properties
+
+The remaining two methods allow you to read or alter the properties of audiorecorder objects.
+
+@DOCSTRING(@audiorecorder/get)
+@DOCSTRING(@audiorecorder/set)
+
 @chapter Audio Processing
 
 Octave provides a few functions for dealing with audio data.  An audio
 `sample' is a single output value from an A/D converter, i.e., a small
 integer number (usually 8 or 16 bits), and audio data is just a series
 of such samples.  It can be characterized by three parameters: the
 sampling rate (measured in samples per second or Hz, e.g., 8000 or
 44100), the number of bits per sample (e.g., 8 or 16), and the number of
